project/src/components/Greeting.js

```diff
@@ -0,0 +1,42 @@
+import React, {Component} from 'react'
+import { connect } from 'react-redux'
+import {withRouter} from 'react-router-dom'
+
+import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
+import '../css/style.css';
+
+const myRouterGreetingComponent = withRouter (class Greeting extends Component {
+
+    render(){
+        const user = this.props.user;
+
+        const onLinkClick = ((e)=>{
```

Comment:
Do not create handlers in the render function here and in all other similar places.